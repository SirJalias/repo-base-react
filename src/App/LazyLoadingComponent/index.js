import React, { useState } from "react";
import VisibilitySensor from "react-visibility-sensor";
import Spinner from "./Spinner";

const LazyLoading = ({ children }) => {
  const [isVisibleComponent, setIsVisible] = useState(false);

  return (
    <VisibilitySensor
      partialVisibility
      active={!isVisibleComponent}
      onChange={isVisible => {
        setIsVisible(isVisible);
      }}
    >
      <div
        style={{
          minHeight: "200px",
          opacity: isVisibleComponent ? 1 : 0,
          transition: "opacity 500ms linear"
        }}
      >
        {isVisibleComponent ? children : <div style={{ height: "1000px" }} />}
      </div>
    </VisibilitySensor>
  );
};

LazyLoading.displayName = "LazyLoading";

export default LazyLoading;
