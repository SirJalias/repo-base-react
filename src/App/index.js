import React from "react";
import styled from "@emotion/styled";
import { Route, Link } from "react-router-dom";
import mq from "lib/mq";
import { useDispatch } from "react-redux";

const Main = styled.div`
  ${mq({
    color: ["red", "green", "blue"]
  })}
`;

function Home() {
  return (
    <Main>
      <h2>Home</h2>
    </Main>
  );
}

function About() {
  return (
    <Main>
      <h2>About</h2>
    </Main>
  );
}

function BasicExample() {
  return (
    <div>
      <ul>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/about">About</Link>
        </li>
        <li>
          <Link to="/topics">Topics</Link>
        </li>
      </ul>

      <hr />

      <Route exact path="/" component={Home} />
      <Route path="/about" component={About} />
    </div>
  );
}

const App = () => {
  const dispatch = useDispatch();
  return (
    <Main>
      This is an app
      <BasicExample />
      <button type="button" onClick={() => dispatch({ type: "TEST" })}>
        TEST
      </button>
    </Main>
  );
};

export default App;
