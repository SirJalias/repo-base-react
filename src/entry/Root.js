import PropTypes from "prop-types";
import React from "react";
import { Provider as ReduxProvider } from "react-redux";
import { ConnectedRouter } from "connected-react-router";
import { createMemoryHistory, createBrowserHistory } from "history";
import configStore from "./store";

const testMode = process.env.NODE_ENV === "test";

const history = testMode ? createMemoryHistory() : createBrowserHistory();

const Root = ({ children, initialStore }) => (
  <ReduxProvider store={configStore(history, initialStore)}>
    <ConnectedRouter history={history}>{children}</ConnectedRouter>
  </ReduxProvider>
);

Root.displayName = "Root";
Root.defaultProps = {
  initialStore: {}
};
Root.propTypes = {
  children: PropTypes.node.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  initialStore: PropTypes.object
};

export default Root;
