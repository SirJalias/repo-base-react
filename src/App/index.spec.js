import React from "react";
import renderer from "react-test-renderer";

import App from "./index";
import Root from "../entry/Root";

let WrapperApp;

beforeEach(() => {
  WrapperApp = () => (
    <Root initialStore={{}}>
      <App />
    </Root>
  );
});

test("Button renders correctly", () => {
  expect(renderer.create(<WrapperApp />).toJSON()).toMatchSnapshot();
});
