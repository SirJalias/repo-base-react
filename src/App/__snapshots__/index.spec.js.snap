// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.emotion-0 {
  color: red;
}

@media (min-width:420px) {
  .emotion-0 {
    color: green;
  }
}

@media (min-width:920px) {
  .emotion-0 {
    color: blue;
  }
}

<div
  className="emotion-0"
>
  This is an app
  <div>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/topics"
          onClick={[Function]}
        >
          Topics
        </a>
      </li>
    </ul>
    <hr />
    <div
      className="emotion-0"
    >
      <h2>
        Home
      </h2>
    </div>
  </div>
  <button
    onClick={[Function]}
    type="button"
  />
</div>
`;
